js/visearch.js

```diff
@@ -122,6 +116,43 @@
     }
   };
 
+  /**
+   * Get query parameters from url [URI] object
+   */
+  function getQueryParamValue(uri, name) {
+    return find([`__vi_${name}`, name], () => uri.getQueryParamValue(name));
+  }
+
+  /**
+   * Wrapper for callback function with additional send result_load event
+   */
+  function callbackWrap(callback, args) {
+    callback(args);
+
+    const curUri = new URI(window.location.href);
+    const reqid = getQueryParamValue(curUri, QUERY_REQID);
+    const imName = getQueryParamValue(curUri, QUERY_IMNAME);
```

Comment:
im_name is for visearch, for Product Search, it should be `pid`

Reply:
the product ID in Recommendations case can be retrieved directly from params in searchbyid, not the URI.
Similarly for the reqid it should be retrieved directly from the response , not the URI. URI is an additional support but lower priority.

I did not see this handled.